packages/pluggableWidgets/tree-view-web/src/components/TreeView.tsx

```diff
@@ -94,22 +126,39 @@ function getTreeViewHeaderAccessibilityProps(isLeafNode: boolean): HTMLAttribute
     };
 }
 
+const treeViewBodyClassName = "widget-tree-view-body";
+
 function TreeViewBranch(props: TreeViewBranchProps): ReactElement {
+    const { level: currentContextLevel } = useContext(TreeViewBranchContext);
     const [treeViewIsExpanded, setTreeViewIsExpanded] = useState<boolean>(props.startExpanded);
+    const [isActualLeafNode, setIsActualLeafNode] = useState<boolean>(props.isUserDefinedLeafNode);
 
     function toggleTreeViewContent(): void {
-        if (!props.isUserDefinedLeafNode) {
+        if (!isActualLeafNode) {
             setTreeViewIsExpanded(isExpanded => !isExpanded);
         }
     }
```

Comment:
This function will be recreated every time a render occurs. I think it's wise to put it inside a useCallback.

Reply:
192fe92